Improve oidc client

m273d15 opened the pull request:
* The [openid connect documentation](http://openid.net/specs/openid-connect-core-1_0.html) recommends to set the state parameter in a authentication request in order to avoid Cross-Site Request Forgery attacks.
* Furthermore there was a small mistake (or legacy part) in the README

m273d15 committed:
Set random oidc client state

* An empty state could allow some sort
  of attacks. Therefore we set a random
  state.

m273d15 committed:
Adjust oidc example README

* rather use docker-compose
  and the .env file to start the example

nils-wisiol commented on oidc_client_example/index_tmp.html:
```diff
@@ -17,10 +17,22 @@
+
+            for (var i = 0; i < length; i++)
+            {
+                result += alphabet[Math.floor(Math.random() * alphabet.length)];
```
`Math.random()` [should not be used for cryptographic purposes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random)

nils-wisiol commented on oidc_client_example/README.md:
```diff
@@ -1,7 +1,8 @@
-1. Build docker image via `docker build -t example_client .`
-2. Run image via `docker run -d -e CLIENT_ID=<client id> -e REDIRECT_URL=<r uri> -e AUTHORITY_URL=<auth. uri> -p 3000:3000 oidc_example`
-Environment variables:
+* Run the client in the project root dir, via `docker-compose -f docker-compose.example.yml up`
```
good catch :)

nils-wisiol reviewed (changes requested):
Please update the random number generator, otherwise  this lgtm

m273d15 committed:
Rather use getRandomValues instead of Math.random()

Using window.crypt.getRandomValues since this method is
recommended for cryptographical purposes.
